erpnext/controllers/subcontracting_controller.py

```diff
@@ -555,6 +567,19 @@ def __validate_supplied_items(self):
 			self.__validate_batch_no(row, key)
 			self.__validate_serial_no(row, key)
 
+	def __set_sourced_by_supplier_items(self):
+		if self.get("sourced_by_supplier_items"):
+			for supplied_item in self.supplied_items:
+				for sourced_by_supplier_item in self.sourced_by_supplier_items:
+					if supplied_item.rm_item_code == sourced_by_supplier_item.get("rm_item_code"):
+						supplied_item.update(
+							{
+								"rate": sourced_by_supplier_item.get("rate"),
+								"amount": sourced_by_supplier_item.get("amount"),
+								"sourced_by_supplier": sourced_by_supplier_item.get("sourced_by_supplier"),
+							}
+						)
```

Comment:
Add a `continue` after the row is updated, so that it doesn't keep looping through the remaining rows